src/ProjectTemplates/BlazorTemplates.Tests/BlazorServerTemplateTest.cs

```diff
@@ -107,84 +136,97 @@ public async Task BlazorServerTemplateWorks_IndividualAuth(bool useLocalDB)
             {
                 Assert.False(
                     aspNetProcess.Process.HasExited,
                     ErrorMessages.GetFailedProcessMessageOrEmpty("Run published project", Project, aspNetProcess.Process));
 
                 await aspNetProcess.AssertStatusCode("/", HttpStatusCode.OK, "text/html");
-                if (BrowserFixture.IsHostAutomationSupported())
+                if (Fixture.BrowserManager.IsAvailable(browserKind))
+                {
+                    var page = await aspNetProcess.VisitInBrowserAsync(browser);
+                    await TestBasicNavigation(page);
+                    await page.CloseAsync();
+                }
+                else
                 {
-                    aspNetProcess.VisitInBrowser(Browser);
-                    TestBasicNavigation();
+                    Assert.False(
+                        TestHelpers.TryValidateBrowserRequired(
+                            browserKind,
+                            isRequired: !Fixture.BrowserManager.IsExplicitlyDisabled(browserKind),
+                            out var errorMessage),
+                        errorMessage);
                 }
             }
         }
 
-        private void TestBasicNavigation()
+        private async Task TestBasicNavigation(IPage page)
         {
-            var retries = 3;
-            var connected = false;
-            do
-            {
-                try
-                {
-                    Browser.Contains("Information: WebSocket connected to",
-                        () => string.Join(Environment.NewLine, Browser.GetBrowserLogs(LogLevel.Info).Select(b => b.Message)));
-                    connected = true;
-                }
-                catch (TimeoutException) when(retries-- > 0)
-                {
-                    Browser.Navigate().Refresh();
-                }
-            } while (!connected && retries > 0);
-
-
-            Browser.Exists(By.TagName("ul"));
+            await page.QuerySelectorAsync("ul");
             // <title> element gets project ID injected into it during template execution
-            Browser.Equal(Project.ProjectName.Trim(), () => Browser.Title.Trim());
+            Assert.Equal(Project.ProjectName.Trim(), (await page.GetTitleAsync()).Trim());
 
             // Initially displays the home page
-            Browser.Equal("Hello, world!", () => Browser.FindElement(By.TagName("h1")).Text);
+            Assert.Equal("Hello, world!", await page.GetTextContentAsync("h1"));
 
             // Can navigate to the counter page
-            Browser.Click(By.PartialLinkText("Counter"));
-            Browser.Contains("counter", () => Browser.Url);
-            Browser.Equal("Counter", () => Browser.FindElement(By.TagName("h1")).Text);
+
+            await Task.WhenAll(
+                page.WaitForNavigationAsync("**/counter"),
+                page.WaitForSelectorAsync("text=Current count: 0"),
+                page.ClickAsync("text=Counter"));
```

Comment:
Just for my education, could you clarify why it's preferable to `WhenAll` here instead of `await`-ing each of them separately?

Reply:
In this case I don't think it matters a lot, I saw this within the playwright docs and thought was the best way to do it. You can totally click counter, await it and then check for the selector.

In some other situations there is an advantage of doing it this way, which is that you can setup a wait condition before you trigger the action, so there is no way that you miss something that happens really fast.